### PR TITLE
[Reply] Make MailView scroll if content no longer fits the screen

### PR DESCRIPTION
**Pull request by Renzo-Olivares**
The current behavior of the `MailViewPage` is to allow the body of the message to scroll if the content does not fit. Instead this change wraps the whole page in a `SingleChildScrollView`, so the whole page scrolls when the content does not fit. This is consistent with the behavior in the Android Reply app. 

* This change also fixes ink splash not appearing on material widgets, by wrapping with `Material()` instead of a `Container()`.
* Uses `SingleChildScrollView` `padding` property instead of a 'Padding' widget.

### CODE REVIEW

File: lib/studies/reply/mail_view_page.dart [hunk 1]
```diff
@@ -18,25 +18,27 @@ class MailViewPage extends StatelessWidget {
...
-            child: Column(
-              crossAxisAlignment: CrossAxisAlignment.start,
-              children: [
-                _MailViewHeader(
-                  email: email,
-                ),
-                const SizedBox(height: 32),
-                Expanded(
-                  child: _MailViewBody(message: email.message),
+        child: SizedBox(
```
Comment (guidezpl) [inline lib/studies/reply/mail_view_page.dart]: This feels a little hacky. I think the better API would be to use `SingleChildScrollView.constraints` or  `IntrinsicHeight`

File: lib/studies/reply/mail_view_page.dart [hunk 2]
```diff
@@ -19,24 +19,24 @@ class MailViewPage extends StatelessWidget {
...
-              children: [
-                _MailViewHeader(
-                  email: email,
-                ),
-                const SizedBox(height: 32),
-                Expanded(
-                  child: _MailViewBody(message: email.message),
-                ),
-              ],
+          height: double.infinity,
```
Comment (rami-a) [inline lib/studies/reply/mail_view_page.dart:22]: Why is this needed?

**Review by rami-a — APPROVED** [review]

File: lib/studies/reply/mail_view_page.dart (hunk 2 above)
Comment (Renzo-Olivares) [reply lib/studies/reply/mail_view_page.dart:22]: It's needed for the `Container` to expand the whole height of the screen. Without it doesn't fill the screen.

File: lib/studies/reply/mail_view_page.dart (hunk 1 above)
Comment (Renzo-Olivares) [reply lib/studies/reply/mail_view_page.dart]: Discussed this in https://github.com/Renzo-Olivares/gallery/pull/22

File: lib/studies/reply/mail_view_page.dart [hunk 3]
```diff
@@ -19,24 +19,24 @@ class MailViewPage extends StatelessWidget {
...
+          child: Material(
+            color: Theme.of(context).cardColor,
```
Comment (johnsonmh) [inline lib/studies/reply/mail_view_page.dart:24]: In general we try to use values from the colorscheme. Would `Theme.of(context).colorScheme.surface` work here?

Comment (johnsonmh) [reply lib/studies/reply/mail_view_page.dart:24]: (apologies because I know this wasn't actually part of your change, only do it if it's no-op!)

**Review by johnsonmh — APPROVED** [review]

File: lib/studies/reply/mail_view_page.dart (hunk 3 above)
Comment (Renzo-Olivares) [reply lib/studies/reply/mail_view_page.dart:24]: `surface` wouldn't work in this case, but I will revisit this in another PR along with other color values that can be inherited from colorScheme.
